src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -611,6 +611,10 @@ private static HaxeClass tryResolveClassByQNameWhenGetQNameFail(@NotNull PsiElem
       String nameWithPackage = getQName(fileChildren, name, true);
       result = findClassByQName(nameWithPackage, type.getContext());
     }
+    else {
+      name = tryResolveFullyQualifiedHaxeReferenceExpression(type);
```

Comment:
This was a bug fixed by @giabao (I think).  If I remember correctly, the basic problem was that the recursion didn't return null properly when it tried looking in the local package.  So, the fix was to split the algorithm up so that the first half would run twice and second half wouldn't run on the second try.